### PR TITLE
Add Lazy component

### LINKED ISSUE #388: Create `Lazy` component
I think we should add a new component: `Lazy`.

This component would be used to lazy load another component. If we decide to start lazy loading Mapbox GL maps, we could include this new component in GLWrapper to handle lazy loading maps, if supported by the browser. I can imagine other components that load a significant amount of data or other packages using this new component to help improve performance.

The component will include:
- The ErrorBoundary component as a fallback if the dynamic import fails
- A loader to display until the component loads

Resources:
- https://reactjs.org/docs/code-splitting.html#reactlazy

### PR DESCRIPTION
**Pull request by katydecorah**
_This PR merges in to the `perf` branch #393 ._

This PR adds a new component, `Lazy`. It's a wrapper for https://loadable-components.com/ which is the React team's recommended alternative for lazy loading components server side (`React.lazy` is not supported on the server side by React, yet). 

`Lazy` accepts two props:

- `lazyComponent` a function to import the desired component
- `lazyHeight` the height of the imported component to reduce content layout shift during load.
- `lazyClasses` is an alternative to `lazyHeight` in case the height of the component is variable, you can define classes to best describe the height of the component.

You must define `lazyHeight` or `lazyClasses`.

You can also pass props to `Lazy` that will be passed along to the import component. The lazy props are named distinctly to avoid prop name collisions.

The component will display a loader while the `lazyComponent` loads. 

Fixes #388 

## How to test

- [ ] Open test cases app (`npm start`) at /Lazy. The example is lazy loading the Search component. Confirm that the Search component loads as expect.

## QA checklist

- [x] No errors logged to console.
- [x] Accessibility score in [Chrome DevTools Audit/Lighthouse](https://developers.google.com/web/tools/lighthouse#devtools) is 100% with Lighthouse version: `6.4.0`.
- [x] Component is accessible at mobile-size viewport.
- [x] Component is accessible at tablet-size viewport.
- [x] Component is accessible at laptop-size viewport.
- [x] Component is accessible at big-monitor-size viewport.

Open the test cases app locally on:

- [x] Chrome, no errors logged to console.
- [x] Firefox, no errors logged to console.
- [x] Safari, no errors logged to console.
- [x] Edge, no errors logged to console.
- [x] IE11, no errors logged to console.
- [x] Mobile Safari, no errors logged to console.
- [x] Android Chrome, no errors logged to console.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -101,6 +101,7 @@
     "@elastic/react-search-ui": "1.3.2",
     "@elastic/react-search-ui-views": "1.3.2",
     "@elastic/search-ui-site-search-connector": "1.3.2",
+    "@loadable/component": "^5.14.1",
```
Comment (katydecorah) [inline package.json:104]: ✏️  Documentation: https://loadable-components.com/

Supports: v5.x (React v16.3+): IE11, IE 9+ (with Map + Set polyfills), all evergreen browsers

**Review by colleenmcginnis — COMMENTED** [review]
@katydecorah is there a good way to slow down loading to see the fallback in action? Or will that be easier to see when we test this out on one of our sites?

**Comment by katydecorah** [comment]
>  is there a good way to slow down loading to see the fallback in action? Or will that be easier to see when we test this out on one of our sites?

@colleenmcginnis it's a little challenging on the test cases app! I just pushed a temporary demo page to the catalog site. Here's a good way to catch the loader: 

1. Pull down this branch
2. Run `npm run build-docs && npm run serve-static` 
3. Open http://localhost:8080/dr-ui/guides/lazy/
4. Open Network tab with the following settings and then hit refresh
image

File: docs/src/pages/guides/lazy.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+order: 5
+layout: page
+contentType: guide
+products:
+  - Documentation
+prependJs:
+  - "import Lazy from '../../../../src/components/lazy/lazy';"
+---
+
+{{<Lazy lazyClasses="h30-mm h36" lazyComponent={() => import('../../../../src/components/search/examples/basic.js')} />}}
```
Comment (katydecorah) [inline docs/src/pages/guides/lazy.md:13]: ✏️ Added this as a testing demo. This page will be the future home of lazy loading documentation -- to be completed in a future PR.

**Comment by katydecorah** [comment]
@colleenmcginnis I just pushed a beta release of a fix for the xtreme test (https://github.com/mapbox/remark-lint-mapbox/pull/59) this should fix the tests! It's good we caught this by making this demo as we'd very likely see this error in all other sites.

**Review by colleenmcginnis — APPROVED** [review]
This looks great to me! I also had fun reading about lazy loading. 🤓 Thanks for the links!

**Pull request merged**